cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -282,43 +295,96 @@ func generateMountsModeRecover(mst *initramfsMountsState) error {
 	}
 
 	// 3. mount ubuntu-data for recovery
+	// start out assuming we have everything and lower expectations as things
+	// fail on us
+	haveHost := true
+	haveSave := true
 	const lockKeysOnFinish = true
 	device, isDecryptDev, err := secbootUnlockVolumeUsingSealedKeyIfEncrypted(disk, "ubuntu-data", boot.InitramfsEncryptionKeyDir, lockKeysOnFinish)
 	if err != nil {
-		return err
+		haveHost = false
+		if isDecryptDev {
+			// the device is a decrypted one, but we failed to unlock it, so we
+			// should try to unlock it with another method(?)
+
+			// TODO:UC20: try another method and reset haveHost = true if it works
+
+			logger.Noticef("failed to unlock encrypted ubuntu-data: %v", err)
+		} else {
+			logger.Noticef("failed to identify ubuntu-data partition to mount: %v", err)
+		}
+		// else no decrypted device and we failed to mount it, so just continue
+		// to recover mode without the host ubuntu-data mount
 	}
 
-	// don't do fsck on the data partition, it could be corrupted
-	if err := doSystemdMount(device, boot.InitramfsHostUbuntuDataDir, nil); err != nil {
-		return err
+	// now mount the host partition (if we have it)
+	saveDir := boot.InitramfsHostWritableDir
+	if haveHost {
+		// don't do fsck on the data partition, it could be corrupted
+		if err := doSystemdMount(device, boot.InitramfsHostUbuntuDataDir, nil); err != nil {
+			logger.Noticef("failed to mount host ubuntu-data: %v", err)
+			haveHost = false
+		}
+	} else {
+		// we don't have the host data, so we should try to mount ubuntu-save
+		// (if it exists) somewhere else
```

Comment:
I don't understand this comment,  we mount save to run, no, here?

Reply:
mmh, I don't think we need this at all. But we need different logic to open save though